agent/tool-scripts/haproxy-ocp

```diff
@@ -0,0 +1,220 @@
+    local tag=$1
+    local haproxy_config=/var/lib/haproxy/conf/haproxy.config
+    local log_dir=$tool_output_dir/logs/$tag
+    mkdir -p $log_dir
+    for router in $(oc get pod --no-headers -n ${router_namespace} | awk '/^router-/ {print $1}') ; do
+        oc exec "${router}" -n ${router_namespace} -- /bin/sh -c "cat ${haproxy_config}" >> $log_dir/${router}-haproxy.config &
+    done
+    wait
+}
+
+router_logs_collect() {
+    local tag=$1
+    local log_dir=$tool_output_dir/logs/$tag
+    mkdir -p $log_dir
+    # watch for reloads
+    oc get pods -n ${router_namespace} | grep '^router-' >> $log_dir/oc_get_pods-router.log &
+    for router in $(oc get pod --no-headers -n ${router_namespace} | awk '/^router-/ {print $1}') ; do
+        oc logs "${router}" -n ${router_namespace} >> $log_dir/${router}.log &
+    done
+    wait
+}
+
+# Process options and arguments
+opts=$(getopt -q -o idp --longoptions "dir:,group:,iteration:,interval:,start,stop,install,postprocess,collect,counters-clear-all,postprocess-script:" -n "getopt.sh" -- "$@")
+
+if [ $? -ne 0 ]; then
+    usage 1
+fi
+
+eval set -- "$opts"
+unset opts
+while true; do
+    case "$1" in
+    --install)
+        mode="install"
+        shift
+    ;;
+    --start)
+        mode="start"
+        shift
+    ;;
+    --stop)
+        mode="stop"
+        shift
+    ;;
+    --postprocess)
+        mode="postprocess"
+        shift
+    ;;
+    --collect)
+        mode="collect"
+        shift
+    ;;
+    --counters-clear-all)
+        counters_clear_all="Y"
+        shift
+    ;;
+    -d|--dir)
+        shift
+        if [ -n "$1" ]; then
+            dir="$1"
+            shift
+        fi
+    ;;
+    -g|--group)
+        shift
+        if [ -n "$1" ]; then
+            group="$1"
+            shift
+        fi
+    ;;
+    -i|--iteration)
+        shift
+        if [ -n "$1" ]; then
+            iteration="$1"
+            shift
+        fi
+    ;;
+    --interval)
+        shift
+        if [ -n "$1" ]; then
+            interval="$1"
+            shift
+        fi
+    ;;
+    --)
+        shift
+        break
+    ;;
+    *)
+        usage 1
+    ;;
+    esac
+done
+
+tool_dir="$dir/tools-$group"
+tool_output_dir=$tool_dir/$tool     # all tools keep data in their tool specific dir
+tool_cmd_file="$tool_output_dir/$tool.cmd"
+tool_pid_file=$pbench_tmp/$group.$iteration.$tool.pid
+tool_stdout_file=$tool_output_dir/$tool-stdout.txt
+tool_stderr_file=$tool_output_dir/$tool-stderr.txt
+
+case "$mode" in
+    install)
+        if [[ -e $ose_openshift_master || -e $origin_master ]]; then
+            check_install_rpm atomic-openshift-clients
+        else
+            printf "This machine is not Openshift master, it is not possible to start $tool on this machine\n"
+            exit 1
+        fi
+    ;;
+    start)
+        mkdir -p $tool_output_dir
+        pushd $tool_output_dir >/dev/null
+        echo "${BASH_SOURCE} --dir=${dir} --interval=${interval} --collect" >$tool_cmd_file
```

Comment:
This was not clear to me (primarily because I had never heard about BASH_SOURCE before, and the "explanation" on the bash man page sucks raw eggs). I would suggest that you use ${BASH_SOURCE[0]} and add a comment to explain what the value is going to be (assuming of course that IIUC).